### PR TITLE
Fix CI to work fine

### PR DESCRIPTION
**Pull request by arukuka**
# Summary

- Change docker image name to `quickmpc-bts`
- Make test steps run all PR, main and develop branch
- Make docker image build run when pushed on main and develop branch

# Purpose

- QuickMPC demo needs BTS Image

# Contents

- Rename workflow file
- Add workflow file for test
- Improve name of docker images, tags

# Testing Methods Performed

- `docker build` for `dev`, `dep-runner` stage

### CODE REVIEW

File: .github/workflows/develop.yml [hunk 1]
```diff
@@ -1,58 +1,31 @@
-name: Build BTS and Run Build and Test in Container
+name: Push nightly BTS Image
```
Comment (mdonaka) [inline .github/workflows/develop.yml]: There are two different notations, `Image` and `Images`.
I'd like to see everything standardized as it's different in several places.

File: .github/workflows/develop.yml [hunk 2]
```diff
@@ -1,58 +1,31 @@
...
       - name: checkout
         uses: actions/checkout@v2
 
-      - name: Build CC image
+      - name: Build BeaverTripleService images
         run: |
-          docker build ../../ --file Dockerfile --target dev --tag ghcr.io/${{ github.repository_owner }}/bts_dev:${{ github.run_id }}
-          docker build ../../ --file Dockerfile --target medium --tag ghcr.io/${{ github.repository_owner }}/bts_medium:${{ github.run_id }}
+          docker build ../../ --file Dockerfile --target dev --tag ghcr.io/${{ github.repository_owner }}/quickmpc-bts:${{ github.run_id }}
+          docker build ../../ --file Dockerfile --target dep-runner --tag ghcr.io/${{ github.repository_owner }}/quickmpc-bts:${{ github.run_id }}-alpine
```
Comment (mdonaka) [inline .github/workflows/develop.yml:19]: Unlike before, the stage name and image name are different, so I need a comment in the readme or dockerfile.

**Review by mdonaka — COMMENTED** [review]
Thanks for the change.
It's approximately LGTM, but I'd like to make some corrections, please.

File: .github/workflows/develop.yml (hunk 1 above)
Comment (arukuka) [reply .github/workflows/develop.yml]: @mdonaka 
Noted with thanks.  
I checked that there are no differences by `grep image -nri .github/`.

File: .github/workflows/develop.yml (hunk 2 above)
Comment (arukuka) [reply .github/workflows/develop.yml:19]: @mdonaka Noted with thanks.
By 2aabc5b23e0de11abd54bb12da643291364bba4f, documentation about docker image and build stage was written in README.md at top directory of repository.

Comment (mdonaka) [reply .github/workflows/develop.yml:19]: I checked it. Very good!

**Review by mdonaka — APPROVED** [review]
LGTM!

**Review by nrkt — APPROVED** [review]
LGTM!!

**Pull request merged**